pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1320,18 +1329,23 @@ func buildHTTPConnectionManager(listenerOpts buildListenerOpts, httpOpts *httpLi
 }
 
 // buildListener builds and initializes a Listener proto based on the provided opts. It does not set any filters.
+// Optionally for HTTP filters with TLS enabled, HTTP/3 can be supported by generating QUIC Mirror filters for the
+// same port (it is fine as QUIC uses UDP)
 func buildListener(opts buildListenerOpts, trafficDirection core.TrafficDirection) *listener.Listener {
 	filterChains := make([]*listener.FilterChain, 0, len(opts.filterChainOpts))
 	listenerFiltersMap := make(map[string]bool)
 	var listenerFilters []*listener.ListenerFilter
 
 	// add a TLS inspector if we need to detect ServerName or ALPN
+	// (this is not applicable for QUIC listeners)
```

Comment:
Because its always inspecting the TLS?

Reply:
Ideally, when UDP and QUIC are multiplexed on the same port, then a listener filter like `QuicInspector` (counterpart of `TlsInspector`) would be needed to decide whether to create `QuicTransportSocket` or `UdpTransportSocket`. However, we are not supporting UDP in Istio at the moment. So it is fine not to have it. That said, I'm not sure if Envoy's `TlsInspector` works with UDP sockets (I guess it does not support DTLS)